packages/apm/src/hubextensions.ts

```diff
@@ -27,6 +27,18 @@ function traceHeaders(this: Hub): { [key: string]: string } {
  * @param context Properties with which the span should be created
  */
 function startSpan(this: Hub, context: SpanContext | TransactionContext): Transaction | Span {
+  /**
+   * @deprecated
+   * We have this here as a fallback to not break users upgrading to the new API
+   */
+  if ((context as any).transaction) {
+    logger.warn(
+      `Please use \`Sentry.startTransaction({name: ${(context as any).transaction}})\` to start a Transaction.`,
+    );
+    (context as TransactionContext).name = (context as any).transaction as string;
+    (context as TransactionContext)._isTransaction = true;
```

Comment:
I may not have fully understood this from previous PRs. What does it mean for a `TransactionContext` to be a Transaction (`_isTransaction = true`)?

Reply:
We have another safeguard using when people call 
`Sentry.startTransaction`

https://github.com/getsentry/sentry-javascript/blob/master/packages/minimal/src/index.ts#L177-L181

Since everything goes through `hub.startSpan` internally with that `_isTransaction` we can have the check when people use JS (without types) if they forget to set the name property

we have this
https://github.com/getsentry/sentry-javascript/blob/master/packages/apm/src/hubextensions.ts#L30-L37